AGENTS.md

```diff
@@ -0,0 +1,433 @@
+# Agent Guide for Yggdrasil Build Scripts
+
+This guide helps AI agents generate correct `build_tarballs.jl` recipes for BinaryBuilder.jl in the Yggdrasil repository.
+
+## Prerequisites
+
+- **BinaryBuilder.jl**: Requires Julia 1.3.0 or later
+- **Supported Platforms**: Linux (glibc and musl for x86_64, i686, aarch64, armv7l, armv6l, ppc64le, riscv64), Windows (x86_64, i686), macOS (x86_64, aarch64), FreeBSD (x86_64, aarch64)
+- Use `supported_platforms()` to get all available platforms
+
+## Essential Structure
+
+Every `build_tarballs.jl` file follows this pattern:
+
+```julia
+using BinaryBuilder
+
+name = "PackageName"              # Valid Julia identifier (no spaces/dashes/dots)
+version = v"X.Y.Z"                # Only major.minor.patch (no prerelease/build tags)
+
+sources = [
+    ArchiveSource("URL", "sha256hash"),
+    # GitSource("URL", "commit_hash"),
+    # DirectorySource("./bundled"),  # for patches
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/package-*
+# Build commands here
+"""
+
+platforms = supported_platforms()  # or filter as needed
+
+products = [
+    LibraryProduct("libname", :libname),
+    ExecutableProduct("exename", :exename),
+]
+
+dependencies = [
+    Dependency("SomeDep_jll"),
+    # BuildDependency("BuildOnlyDep_jll"),
+]
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+```
+
+## Critical Rules
+
+### Naming
+
+- **Name**: Must be a valid Julia identifier. Replace spaces/dashes with underscores. Match upstream casing.
+- **Version**: Only `X.Y.Z` format. Truncate any `-alpha`, `+build`, or 4+ level versions.
+- **Products**: Export symbols match the library/executable names (as symbols: `:libname`).
```

Comment:
This isn't a hard rule actually, just general guidance. It may be that it makes more sense to name it something else.